mmdet/datasets/coco.py

```diff
@@ -383,19 +383,22 @@ def format_results(self, results, jsonfile_prefix=None, **kwargs):
         result_files = self.results2json(results, jsonfile_prefix)
         return result_files, tmp_dir
 
-    def evaluate(self,
-                 results,
-                 metric='bbox',
-                 logger=None,
-                 jsonfile_prefix=None,
-                 classwise=False,
-                 proposal_nums=(100, 300, 1000),
-                 iou_thrs=None,
-                 metric_items=None):
-        """Evaluation in COCO protocol.
+    def evaluate_ins_json(self,
+                          results,
+                          result_files,
```

Comment:
If this function is named "evaluate json", why the raw results are still needed? This function should only process the json file.

Reply:
That's right. Function name should be modified. `results` is only for metric `proposal fast`. `result_files` is for other metric like 'segm', 'bbox'.